### PR TITLE
PR to fix ansible-ai-connect-service: Cross-site Scripting (XSS) in serialize-javascript

### PR DESCRIPTION
**Pull request by justjais**
Jira Issue: <https://issues.redhat.com/browse/AAP-31381>

## Description

PR to fix ansible-ai-connect-service: Cross-site Scripting (XSS) in serialize-javascript

### Steps to test
1. Pull down the PR
2. Run `npm install` and node packages should install as expected w/o failures
3. Run `npm audit fix` to verify if there's any vulnerabilities, currently there's 0.

### Scenarios tested

NA

## Production deployment

- [x] This code change is ready for production on its own
- [ ] This code change requires the following considerations before going to production:

### CODE REVIEW

**Comment by jameswnl** [comment]
we probably have to fix it in the `package.json`?

**Comment by manstis** [comment]
Upgrade to PF5 isn't easy.

You'll need to bump the version of Ansible UI framework too (to one using PF5) and fix the fallout too.

**Comment by justjais** [comment]
@manstis yea, this isn't a 1 story point and straight CVE fix, for fixing this patternfly needs to be updated to ^5.1 coz npm update updates ansible-ui-framework package which is in turn dependent over patternfly ^5.1, and the fallout needs to be resolved as you mentioned.

I'll bring this up during today's scrum and how to move forward with the issue.

File: ansible_ai_connect_admin_portal/src/SingleInlineEdit.tsx [hunk 1]
```diff
@@ -24,7 +24,8 @@ export const SingleInlineEdit = (props: InlineTextInputProps) => {
         <InputGroup>
           <TextInput
             type={isPassword && passwordHidden ? "password" : "text"}
-            onChange={(value, event) => props.onChange?.(value)}
+            onChange={(_event, value: string) => onChange(value)}
+            // onChange={(value, event) => props.onChange?.(value)}
```
Comment (manstis) [inline ansible_ai_connect_admin_portal/src/SingleInlineEdit.tsx:28]: Do we need this comment?

File: ansible_ai_connect_admin_portal/src/__tests__/AppHeader.test.tsx [hunk 2]
```diff
@@ -20,26 +20,16 @@ describe("AppHeader", () => {
...
+        <AppHeader userName={"Batman"} />
+      </BrowserRouter>
+    );
+    const accountMenu = await screen.findByTestId("page-masthead-dropdown__button");
     expect(accountMenu).toBeInTheDocument();
     expect(accountMenu).toHaveTextContent("Batman");
 
     // Check "Logout" option is not present
     expect(screen.queryByText("Logout")).toBeNull();
-
```
Comment (manstis) [inline ansible_ai_connect_admin_portal/src/__tests__/AppHeader.test.tsx:30]: You seem to have remove the test for logging out?

**Review by manstis — CHANGES_REQUESTED** [review]
The styling is now, unfortunately, wrong:

**AAP-31381**
AAP-31381

**Original**
original

Furthermore the "kebab" menu (top left corner) does not work properly.

It should collapse the left-hand-side navigation menu.

**Review by manstis — APPROVED** [review]
The icon/text alignment in the left-hand-side menu still is not correct; but the kebab menu works now :smile_cat: 

image

Approving; but I suggest a follow-up JIRA to fix the alignment be made.

**Comment by justjais** [comment]
@manstis thanks for the review, and giving green light to the PR. Also, I've opened a bug jira: https://issues.redhat.com/browse/AAP-36777, to address the alignment issue.

**Pull request merged**